src/components/Form.js

```diff
@@ -26,7 +26,7 @@ export default class Form extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    this.setState(this.getStateFromProps(nextProps));
+    this.state = {...this.state, ...this.getStateFromProps(nextProps)};
```

Comment:
At first I was frightened by this change as it's always been documented as a bad practice. But now I've read more about the lifecycle and the execution order of hooks, I it feels like we don't have many other alternatives. We may want to drop a comment here to explain why we're doing this, though.